cpmpy/expressions/globalconstraints.py

```diff
@@ -417,6 +417,37 @@ def get_bounds(self):
         bnds = [get_bounds(x) for x in self.args]
         return max(lb for lb,ub in bnds), max(ub for lb,ub in bnds)
 
+class Isin(GlobalConstraint):
+    """
+        The "isin" constraint, defining non-interval domains for an expression
+    """
+
+    def __init__(self, expr, arr):
+        self.expr = expr
```

Comment:
We typically don't store the individual arguments in the object.
If you need it later, you can simply do:
```python
expr, arr = self.args
````